lib/orbit-common/cache-transformer.js

```diff
@@ -0,0 +1,92 @@
+import { eq } from 'orbit/lib/eq';
+import { Class } from 'orbit/lib/objects';
+
+export default Class.extend({
+  init: function(retrieve, transformTarget, schema, processors){
```

Comment:
In the case of orbit-firebase the operations aren't applied directly to the document, so it's necessary to send them to something else. I see a few options for this:

1) Return the complete collection of operations:

    docTransformationBuilder.build(operation) => [beforeOperation, operation, afterOperation...]

2) Emit the operations:

     transformer.on('operation', function(operation){ ... }); // each individual operation emitted in series
     transformer.process(originalOperation);

3) Pass in the target to receive the operations:

    transformer = new Transformer(... target) // where target has a target.transform(operation) method
    transformer.process(originalOperation);